src/shared_utilities/models.jl

```diff
@@ -404,6 +404,15 @@ function get_drivers(model::AbstractModel)
     return (nothing, nothing)
 end
 
+"""
+    add_dfluxBCdY_to_cache(p::NamedTuple, model::AbstractModel, _, _)
+
+We only need to add this field in the case of a `RichardsModel`, so we
+provide a fallback method for all other models which does nothing.
+"""
+function add_dfluxBCdY_to_cache(p::NamedTuple, model::AbstractModel, _, _)
```

Comment:
I dont think we need this because we have the `boundary_vars` (and types, domain_names) functions which have the capability to add these variables already.